src/components/Main.js

```diff
@@ -29,32 +34,70 @@ export default function Main() {
   const [coinList, setCoinList] = useState([]);
   const [searchCoin, setSearchCoin] = useState("");
 
+  const dispatch = useDispatch();
+
+  const tickerCoinList = useSelector((state) => state.socket.coinList);
+  const realTimeCoin = useSelector((state) => state.socket.socketCoin);
+
+  useEffect(() => {
+    const myCoin = "ALL";
+
+    const fetchCoinList = function () {
+      dispatch(requestCoinList(myCoin));
+    };
+
+    fetchCoinList();
+  }, []);
+
+  useEffect(() => {
+    const tickerCoin = JSON.stringify(tickerCoinList);
+    const parseTickerCoin = JSON.parse(tickerCoin);
+    const coinName = Object.keys(parseTickerCoin);
+    const coinInfo = Object.values(parseTickerCoin);
+
+    coinName.pop();
+    coinInfo.pop();
+
+    for (let i = 0; i < coinInfo.length; i++) {
+      coinInfo[i].currency_name = coinName[i];
+    }
+
+    setCoinList(coinInfo);
+  }, [tickerCoinList]);
+
   useEffect(() => {
     ws.onmessage = (event) => {
       const res = JSON.parse(event.data);
-      const coinName = Object.keys(res.data);
-      const coinInfo = Object.values(res.data);
-      coinName.pop();
-      coinInfo.pop();
-
-      for (let i = 0; i < coinInfo.length; i++) {
-        coinInfo[i].currency_name = coinName[i];
-      }
 
-      setCoinList(coinInfo);
+      dispatch(requestSocketData(res.content));
       ws.send("클라이언트에서 서버로 답장을 보냅니다.");
     };
 
     ws.onerror = (error) => {
       console.error(error);
     };
+
+    return () => {
+      ws.close();
+    };
   }, []);
 
   const filteredCoinList =
     searchCoin === ""
       ? coinList
       : coinList.filter((coin) => coin.currency_name === searchCoin);
 
+  coinList.forEach((coin) => {
+    if (realTimeCoin.symbol) {
+      if (coin.currency_name === realTimeCoin.symbol.slice(0, 3)) {
+        coin.closing_price = realTimeCoin.closePrice;
+        coin.fluctate_rate_24H = realTimeCoin.chgRate;
+        coin.acc_trade_value_24H = realTimeCoin.value;
+      }
+      return coin;
```

Comment:
```suggestion
      }
      
      return coin;
```
한 칸 개행해주시면 가독성이 조금 더 좋아질 거 같습니다 !

Reply:
적용했습니다!